benchmarks/cpp/nvfuser/softmax.cpp

```diff
@@ -97,62 +141,111 @@ static void Softmax_Dropout(benchmark::State& benchmark_state) {
-  fusion.addOutput(dropout_results.mask);
+  fusion->addOutput(dropout_results.mask);
+}
+
+static void nvFuserScheduler_SoftmaxDropout(
+    benchmark::State& benchmark_state,
+    FusionExecutorCache* fusion_executor_cache,
+    DataType dtype,
+    const int kReductionAxis) {
+  TORCH_INTERNAL_ASSERT(dtype == DataType::Float || dtype == DataType::Half);
+
+  // reduce across 1, [256, 12, 100, 8]
+  std::vector<int64_t> input_shape{256, 12, 100, benchmark_state.range(0)};
+
+  constexpr int kHiddenSize = 768;
+  constexpr int kNumAttentionHeads = 12;
+  constexpr int kAttentionHeadSize = kHiddenSize / kNumAttentionHeads;
+  constexpr float kDropoutProbability = 0.9;
+  constexpr float kScale = 1.0f / kDropoutProbability;
 
   // inputs
   at::manual_seed(0);
-  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  auto options =
+      at::TensorOptions().dtype(data_type_to_aten(dtype)).device(at::kCUDA, 0);
   at::Tensor at_scores = at::randn(input_shape, options);
   at::Tensor at_mask = at::randn(input_shape, options);
-  std::vector<c10::IValue> inputs(
+  std::vector<c10::IValue> aten_inputs(
       {at_scores, at_mask, sqrt(kAttentionHeadSize)});
 
-  // outputs
-  std::vector<at::Tensor> outputs;
+  fusion_executor_cache->profile(true);
+  fusion_executor_cache->runFusionWithInputs(aten_inputs);
 
-  auto reduction_params = getNormalizationHeuristics(&fusion, inputs);
-  TORCH_CHECK(reduction_params, "Reduction schedule was not generated!");
+  auto compile_log = fusion_executor_cache->getMostRecentExecutorInfo();
+  auto executor_instance = compile_log.fusion_executor;
+  TORCH_INTERNAL_ASSERT(compile_log.reduction_params.has_value());
+  TORCH_INTERNAL_ASSERT(compile_log.launch_constraints.has_value());
+  auto rparams = toString(compile_log.reduction_params.value());
+  auto lparams = toString(compile_log.launch_constraints.value());
 
-  scheduleNormalization(&fusion, reduction_params.value());
-
-  FusionExecutor executor;
-  executor.setMeasureKernelTimeFlag(true);
-  executor.compileFusion(&fusion);
+  benchmark_state.SetLabel(rparams + lparams);
 
+  fusion_executor_cache->profile(false);
+  executor_instance->setMeasureKernelTimeFlag(true);
+  // Sync everything up before we start
   cudaDeviceSynchronize();
   for (auto _ : benchmark_state) {
-    outputs = executor.runFusion(
-        c10::ArrayRef<c10::IValue>(inputs), reduction_params.value().lparams);
-    benchmark_state.SetIterationTime(executor.kernelTimeMs() / 1000.0);
-    cudaDeviceSynchronize();
+    auto cg_outputs = fusion_executor_cache->runFusionWithInputs(aten_inputs);
+    benchmark_state.SetIterationTime(
+        executor_instance->kernelTimeMs() / 1000.0);
   }
+  // Sync everything up before we're finished, don't want to run ahead on the
+  // cpu while benchmarking.
+  cudaDeviceSynchronize();
+
+  benchmark_state.SetBytesProcessed(
+      int64_t(benchmark_state.iterations()) * 6 * 256 * 12 * 100 *
```

Comment:
Should `SetBytesProcessed` be `5 * kInputNumel * dataTypeSize(dtype) + kInputNumel * dataTypeSize(Bool)` where `kInputNumel = 256 * 12 * 100 * benchmark_state.range(0)`?